vimoutliner/scripts/votl_maketags.py

```diff
@@ -0,0 +1,119 @@
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program; if not, see <http://www.gnu.org/licenses/>.
+from __future__ import print_function
+import os.path
+import os
+import re
+import sys
+import subprocess
+import shutil
+import tempfile
+
+TAGFILENAME = os.path.expanduser("~/.vim/vimoutliner/vo_tags.tag")
+
+
+def makeTagFileStartingAt(args):
+    if len(args) != 1:
+        print("""
+        Usage is:
+            otltags topLevelOutlineFileName""")
+        sys.exit(1)
+
+    absoluteFileName = os.path.abspath(args[0])
+
+    processedFiles = {}
+    recordFileAsProcessed(absoluteFileName, processedFiles)
+    os.unlink(TAGFILENAME)
+    process1Outline(absoluteFileName, processedFiles)
+    sortAndDeleteDupsFromTagFile(absoluteFileName)
+
+
+def sortAndDeleteDupsFromTagFile(infile):
+    with tempfile.TemporaryFile() as temptagfile:
+        proc = subprocess.Popen(['sort', '-u'], stdin=infile,
+                                stdout=temptagfile)
+        proc.wait()
+        shutil.copyfileobj(temptagfile, open(infile, 'w'))
+
+
+def process1Outline(fileName, processedFilesHashRef):
+    if not os.path.exists(fileName):
+        makeDirectory(fileName)
+        makeEmptyFile(fileName)
+
+    print("Begin processing file {}.".format(fileName))
+
+    baseDirectory = os.path.dirname(fileName)
+    tags = getTagsFromFile(fileName)
+    for tagKey in tags:
+        if tags[tagKey].startswith('/'):
+            absoluteFileName = tags[tagKey]
+        else:
+            absoluteFileName = os.path.abspath(os.path.join(baseDirectory,
+                                                            tags[tagKey]))
+        appendTagToTagFile(tagKey, absoluteFileName)
+
+        if notProcessedYet(absoluteFileName, processedFilesHashRef):
+            recordFileAsProcessed(absoluteFileName, processedFilesHashRef)
+            process1Outline(absoluteFileName, processedFilesHashRef)
+
+
+def appendTagToTagFile(tagline, filename):
+    with open(filename, 'w') as tagfile:
+        print("{0}        {1}        :1".format(tagline), file=tagfile)
+
+
+def makeEmptyFile(filename):
+    with open(filename, 'w') as outlinefile:
+        outlinefile.write('')
+
+
+def makeDirectory(completeFileName):
+    directoryName = os.path.dirname(completeFileName)
+    os.makedirs(directoryName)
+
+
+def getTagsFromFile(outlineFileName):
+    tags = {}
+
+    tagString = ""
+    with open(outlineFileName, 'r') as outlinefile:
+        for line in outlinefile:
+            line = line.strip()
+
+            if re.search(r'^\s*(_tag_\S+)', line):
+                tagString = 1
```

Comment:
`1` is incorrect. What `$1` in Perl does is get the contents of the first capture group. What you want is

```python
matches = re.search(r'^\s*(_tag_\S+)', line)
if m:
    tagString = m.group(1)
```